File: ZeroDay/app/src/main/java/com/example/zeroday/views/MainActivity.java
package com.example.zeroday.views;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.example.zeroday.R;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //On récupère la barre de navigation
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);

        //On séléctionne l'item "chart" par défaut
        bottomNavigationView.setSelectedItemId(R.id.chart);
        //On charge le frament correspondant à l'item "chart" par défaut
        loadFragment( new ChartFragment());

        //On définit le OnItemSelectedListener de façon à charger le bon fragment en fonction de l'item selectionné
        bottomNavigationView.setOnItemSelectedListener(item -> {

            switch (item.getItemId()){

                case R.id.chart:
                    loadFragment(new ChartFragment());
                    break;
                case R.id.add:
                    loadFragment(new AddFragment());
                    break;
                case R.id.piggy:
                    loadFragment(new PiggyFragment());
                    break;
            }
            return true;
        });

    }

    //Fonction permettant de charger le fragment passé en paramètres dans le conteneur central de l'activité
    private void loadFragment(Fragment fragment){
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.fragment_container,fragment);
        transaction.disallowAddToBackStack();
        transaction.commit();
    }
}

File: ZeroDay/app/src/main/java/com/example/zeroday/views/MainActivity.kt
package com.example.zeroday.views

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.example.zeroday.R
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val navigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation)

        navigationView.setSelectedItemId(R.id.add)
        loadFragment(AddFragment())

        navigationView.setOnItemSelectedListener {
            when (it.itemId) {
                R.id.chart -> {
                    loadFragment(ChartFragment())
                    return@setOnItemSelectedListener true
                }
                R.id.add -> {
                    loadFragment(AddFragment())
                    return@setOnItemSelectedListener true
                }
                R.id.piggy -> {
                    loadFragment(PiggyFragment())
                    return@setOnItemSelectedListener true
                }
                else -> false
            }
        }
    }
    //Fonction de chargement de fragment
    private fun loadFragment(fragment : Fragment){
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.fragment_container,fragment)
        transaction.disallowAddToBackStack()
        transaction.commit()
    }
}